go-controller/pkg/node/gateway_init.go

```diff
@@ -178,6 +178,22 @@ func getGatewayNextHops() ([]net.IP, string, error) {
 			}
 		}
 	}
+
+	if config.Gateway.Mode == config.GatewayModeLocal {
```

Comment:
in this commit, this code isn't in the right spot. I think you fixed in the 2nd commit, but really you should fix it here.